shared/src/main/java/org/geysermc/hydraulic/pack/PackManager.java

```diff
@@ -157,4 +208,108 @@ private void callEvent(@NotNull ModInfo mod, @NotNull Event event) {
             module.call(event.getClass(), new PackEventContext(this.hydraulic, mod, module, event));
         }
     }
+
+    private void initializeModLookups() {
+        // Step 1: Lookup which namespaces are contained by which mods
+        final Multimap<String, ModInfo> namespacesToMods = this.namespacesToMods;
+        namespacesToMods.clear();
+        for (final ModInfo mod : hydraulic.mods()) {
+            for (final Path root : mod.roots()) {
+                final Path assets = root.resolve("assets");
+                if (!Files.isDirectory(assets)) continue;
+                try (Stream<Path> stream = Files.list(assets)) {
+                    stream.filter(Files::isDirectory)
+                        .map(Path::getFileName)
+                        .map(Path::toString)
+                        .filter(namespace -> !namespace.equals("minecraft"))
+                        .forEach(namespace -> namespacesToMods.put(namespace, mod));
```

Comment:
Will this not break if multiple mods have the same namespace? If so maybe just add a comment if its unlikely we can easily fix that

Reply:
No, it won't break. That's what a `Multimap` is. `ListMultimap<K, V>` is basically a `Map<K, List<V>>`. See Guava's wiki on [multimaps](https://github.com/google/guava/wiki/NewCollectionTypesExplained#multimap).

Reply:
Gotcha missed the declaration